src/node_crypto.cc

```diff
@@ -741,6 +743,64 @@ void SecureContext::AddCRL(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void SecureContext::UseExtraCaCerts(const char* fname) {
+  extra_root_certs_file = fname;
+}
+
+static void emitWarning(Environment* env, const char* fmt, ...) {
+  char warning[1024];
+  va_list ap;
+
+  va_start(ap, fmt);
+  vsnprintf(warning, sizeof(warning), fmt, ap);
+  va_end(ap);
+
+#if 0
+  Local<Object> process = env->process_object();
+  Local<v8::Value> f = process->Get(
+    FIXED_ONE_BYTE_STRING(env->isolate(), "emitWarning"));
+  Local<Value> args[] = {
+    node::OneByteString(env->isolate(), warning, strlen(warning))
+  };
+  f.As<v8::Function>()->Call(process, arraysize(args), args);
+
+#else
+  Local<Object> process = env->process_object();
+  Local<Value> args[] = {
+    node::OneByteString(env->isolate(), warning, strlen(warning))
+  };
+  MakeCallback(env, process, "emitWarning", arraysize(args), args);
+#endif
+}
+
+static void LoadExtraCaCerts(Environment* env) {
+  if (!extra_root_certs_file)
+    return;
+
+  BIO *bio = BIO_new_file(extra_root_certs_file, "r");
+  if (!bio) {
+    unsigned long err = ERR_get_error();  // NOLINT(runtime/int)
+    emitWarning(env, "Ignoring extra certs from `%s`, load failed: %s\n",
+            extra_root_certs_file, ERR_error_string(err, NULL));
+    return;
+  }
+  while (X509* x509 =
+      PEM_read_bio_X509(bio, nullptr, CryptoPemCallback, nullptr)) {
+    X509_STORE_add_cert(root_cert_store, x509);
+    X509_free(x509);
+  }
+  BIO_free_all(bio);
+
+  unsigned long err = ERR_peek_error();  // NOLINT(runtime/int)
+  // Clear error if its EOF/no start line found.
+  if (ERR_GET_LIB(err) == ERR_LIB_PEM &&
+      ERR_GET_REASON(err) == PEM_R_NO_START_LINE) {
+    ERR_clear_error();
+  } else if (err) {
+    emitWarning(env, "Ignoring extra certs from `%s`, load failed: %s\n",
+            extra_root_certs_file, ERR_error_string(err, NULL));
+  }
+}
```

Comment:
This function could share code with SSL_CTX_use_certificate_chain.  Also, you could add a `MarkPopErrorOnReturn mark_pop_error_on_return` at the top so you don't have to manually juggle the error stack.